controllers/topology/reconcile_state_test.go

```diff
@@ -563,6 +563,55 @@ func TestReconcileRepeatedly(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Should drop map key (outside of .spec.machineTemplate.metadata)",
+			reconcileSteps: []interface{}{
+				reconcileStep{
+					name: "Initially reconcile KCP",
+					desired: object{
+						annotations: map[string]interface{}{},
+						spec: map[string]interface{}{
+							"someMap": map[string]interface{}{
+								"map.key.with.dots/owned": "true",
+								"anotherLabel":            "true",
+							},
+						},
+					},
+					want: object{
+						annotations: map[string]interface{}{
+							"topology.cluster.x-k8s.io/managed-fields": "someMap.anotherLabel, someMap.map.key.with.dots/owned",
+						},
+						spec: map[string]interface{}{
+							"someMap": map[string]interface{}{
+								"map.key.with.dots/owned": "true",
+								"anotherLabel":            "true",
+							},
+						},
+					},
+				},
+				reconcileStep{
+					name: "Drop the map key with dots",
+					desired: object{
+						annotations: map[string]interface{}{},
+						spec: map[string]interface{}{
+							"someMap": map[string]interface{}{
+								"anotherLabel": "true",
+							},
+						},
+					},
+					want: object{
+						annotations: map[string]interface{}{
+							"topology.cluster.x-k8s.io/managed-fields": "someMap.anotherLabel",
+						},
+						spec: map[string]interface{}{
+							"someMap": map[string]interface{}{
+								"anotherLabel":            "true",
```

Comment:
This test currently fails because `"map.key.with.dots/owned": "true",` is not deleted.